common/src/api/external/mod.rs

```diff
@@ -2214,320 +2214,6 @@ pub struct LoopbackAddress {
     pub address: IpNet,
 }
 
-/// A switch port represents a physical external port on a rack switch.
```

Comment:
I'm okay with moving this stuff into `types/src/external_api`, but I'd caution -- if we're going to do it for all types, we should plan on doing it pretty soon. Omicron suffers a bit from the syndrome of "we should do a refactor, but it's big, so we only got halfway through, and now things are a little more spread out than they were before".

However, if the goal is to chunk this up into submodules where things are related, I'm totally on-board, that seems like a win.

Reply:
Hopefully this one is easier because it's _mostly_ grouping things within each crate. Last time I think the biggest difficulty I ran into was some types I couldn't untangle to move out of `omicron-common`.

Reply:
My hope for this is--if we like this direction--to as quickly as possible, dogpile this migration